gatorgrouper/utils/group_graph.py

```diff
@@ -68,30 +69,34 @@ def group_graph_partition(
         for j, w2 in enumerate(weights[:i]):
             score = compatibility(w1, w2)
             G.add_edge(i, j, weight=score)
-            print(i, j, w1, w2, score)
 
     # Partition the vertices
     partition = recursive_kl(G, numgrp=numgrp)
     groups = []
-    print("Partition:")
     for p in partition:
         groups.append([responses[i] for i in p])
-        print([G.nodes[i]["weight"] for i in p])
-    print("Total cut size:", total_cut_size(G, partition))
-    return groups
+    output = list()
+    # Generate a list of list of list in order to fit the required display input
+    for student in groups:
+        temp = list()
+        for value in student:
+            value += " fakestring"
```

Comment:
`" fakestring"` looks like a debugging artefact. What is its purpose? If a simple string is needed to hold a place, you should use an empty string, or `None`.

Reply:
As I mentioned in issue #241, this code is due to that I need to make the output of this function to be the list of the list of the list. I think there should be a way to do this better than what I did, and that should be handled by the next pull request.

Reply:
@Lancasterwu I believe that if you change `responses[i]` to `inputlist[i]` on line 118 and then `return groups` it should do what you want to it

Reply:
The `inputlist[i]` is a list which contains the student's name at position 0, and there are many weights numbers follow that name. The partition is assigned to the students' name, If we put weights also to those position based on partition, wouldn't that make the output that we don't want? I may have some misunderstanding of the code here.

Reply:
My understanding was that `inputlist` is a list containing each parsed row of the CSV data (where [1:] become the weights), and that you wanted the final output to be a list of lists of lists, with the the form
```
[
# list of groups
    [
    # first group
        [
        # CSV row corresponding to first student in group
            "student name", True, 0.5, "some other field in data", False, ...
        ]
    ]
]
```
If that's the right format, then my above suggestion should apply, since we need to use the index contained in the partition in order to lookup the entire CSV row rather than the student name in the first position of the row.

Reply:
This is what I thought the structure of the `inputlist` should be. Then I think I got confused about the `partition` part. I have tried and it works. Thanks for the suggestion!